test/test_host.go

```diff
@@ -73,26 +78,114 @@ func (s *HostSuite) TestNetworkedPersistentJob(t *c.C) {
 			Env: map[string]string{
 				"NAME": serviceName,
 			},
-		},
+		}.Merge(extraConfig),
 	})
-	cmd.Start()
-	defer cmd.Kill()
+	cmd.HostID = h.ID()
+	if err := cmd.Start(); err != nil {
+		return nil, nil, err
+	}
 
-	// get the ip:port that that job exposed.
-	// phone discoverd and ask by serviceName -- we set a unique one so this works with concurrent tests.
-	instances, err := s.discoverdClient(t).Instances(serviceName, 15*time.Second)
-	t.Assert(err, c.IsNil)
-	t.Assert(instances, c.HasLen, 1)
+	// wait for the job to heartbeat and return its address
+	services, err := dc.Instances(serviceName, time.Second*100)
+	if err != nil {
+		cmd.Kill()
+		return nil, nil, err
+	}
+	if len(services) != 1 {
+		cmd.Kill()
+		return nil, nil, fmt.Errorf("test setup: expected exactly one service instance, got %d", len(services))
+	}
+
+	return cmd, services[0], nil
+}
 
+func runIshCommand(service *discoverd.Instance, cmd string) (string, error) {
 	resp, err := http.Post(
-		fmt.Sprintf("http://%s/ish", instances[0].Addr),
+		fmt.Sprintf("http://%s/ish", service.Addr),
 		"text/plain",
-		strings.NewReader("echo echocococo"),
+		strings.NewReader(cmd),
 	)
-	t.Assert(err, c.IsNil)
-	defer resp.Body.Close()
+	if err != nil {
+		return "", err
+	}
 	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+func (s *HostSuite) TestNetworkedPersistentJob(t *c.C) {
+	// this isn't much more impressive than what's already running by the time we've got a cluster engaged
+	// but the idea is to use this basic design to enable testing a series manipulations on a single container.
+
+	cluster := s.clusterClient(t)
+
+	// run a job that accepts tcp connections and performs tasks we ask of it in its container
+	cmd, service, err := makeIshApp(cluster, s.anyHostClient(t), s.discoverdClient(t), host.ContainerConfig{})
 	t.Assert(err, c.IsNil)
+	defer cmd.Kill()
+
+	// test that we can interact with that job
+	resp, err := runIshCommand(service, "echo echocococo")
+	t.Assert(err, c.IsNil)
+	t.Assert(resp, c.Equals, "echocococo\n")
+}
+
+func (s *HostSuite) TestVolumeCreation(t *c.C) {
+	h := s.anyHostClient(t)
+
+	vol, err := h.CreateVolume("default")
+	t.Assert(err, c.IsNil)
+	t.Assert(vol.ID, c.Not(c.Equals), "")
+}
+
+func (s *HostSuite) TestVolumeCreationFailsForNonexistantProvider(t *c.C) {
+	h := s.anyHostClient(t)
+
+	vol, err := h.CreateVolume("non-existent")
+	t.Assert(err, c.NotNil)
+	t.Assert(vol.ID, c.Equals, "")
```

Comment:
This assert shouldn't happen, as you're not guaranteed that the first param is not nil if there is a non-nil err return.